library_of_babel.py

```diff
@@ -30,7 +30,7 @@
 
 
 def text_prep(text):
-    digs = 'abcdefghijklmnopqrstuvwxyz, .'
+    digs = set('abcdefghijklmnopqrstuvwxyz, .')
```

Comment:
digs.index will not work anymore, as sets have no index. Since you are seeing if letter.lower() is in digs, just do
`prepared += letter.lower()` on line 39